stan/math/rev/prob/std_normal_log_qf.hpp

```diff
@@ -19,16 +19,14 @@ namespace math {
  */
 template <typename T, require_stan_scalar_or_eigen_t<T>* = nullptr>
 inline auto std_normal_log_qf(const var_value<T>& log_p) {
-  return make_callback_var(
-      std_normal_log_qf(log_p.val()), [log_p](auto& vi) mutable {
-        auto vi_array = as_array_or_scalar(vi.val());
-        auto vi_sign = sign(as_array_or_scalar(vi.adj()));
-
-        const auto& deriv = as_array_or_scalar(log_p).val()
-                            + log(as_array_or_scalar(vi.adj()) * vi_sign)
-                            - NEG_LOG_SQRT_TWO_PI + 0.5 * square(vi_array);
-        as_array_or_scalar(log_p).adj() += vi_sign * exp(deriv);
-      });
+  const auto& arena_rtn = to_arena(std_normal_log_qf(log_p.val()));
+  return make_callback_var(arena_rtn, [log_p, arena_rtn](auto& vi) mutable {
+    auto deriv = apply_scalar_binary(
+        log_p.val(), arena_rtn, [](const auto& logp_val, const auto& rtn_val) {
+          return exp(logp_val - std_normal_lpdf(rtn_val));
+        });
```

Comment:
Instead of `apply_scalar_binary` here how do you feel about just doing the `std_normal_lpdf` on the return val? Since then the rest can use standard vectorization from Eigen.

```suggestion
    auto derive = exp(log_p.val() -  arena_rtn.unaryExpr([](auto x) { return std_normal_lpdf(x);}));
```

Same thing for the other change.